wgsl/index.bs

```diff
@@ -5340,6 +5340,145 @@ Issue: [[#uniform-control-flow]] needs to state whether all invocations being di
   </xmp>
 </div>
 
+### Control Flow Interruption ### {#control-flow-interruption-section}
+
+A <dfn>control flow interruption</dfn> is any one of the following statements:
+* a [=statement/break=] statement
+* a [=statement/continue=] statement
+* a [=statement/fallthrough=] statement
+* a [=statement/return=] statement
+* a [=statement/discard=] statement
+* a [=compound statement=], if one of its members is a [=control flow interruption=].
+
+A control flow interruption must:
+* be a [=continue=] statement immediately followed by a [=continuing=] statement, or
+* appear last in its immediately enclosing compound statement.
```

Comment:
Does this need a better definition? The use of immediate makes me think of dominators. When I read the bit below that control flow statements are not compound statements, I'm left wondering whether they are skipped or stop consideration. For example:
```
...
// Is this outer compound statement the immediately enclosing compound statement 
// of the compound statement in the if or is there none?
{
  if (c) {
    return;
  }
  a = 5;
}
...
```

I think the examples below clarify this case, but I find the wording potentially confusing.

Reply:
There's a paragraph just past the subsequent note that tries to address this;

> Switch, if, loop, and for statements contain compound statements, but are not themselves compound statements. (That is, they do not match the compound_statement grammar rule.) They are never classified as control flow interruptions.

The "if" and "loop" statements are phrased in terms of conditionally or repeatedly executing a compound statement (with a cross-reference).  Similar for the "continuing" statement.

On the other hand: 

I just noticed that the a case_body is not defined in terms of compound_statement.  It's a special case construction to ensure the fallthrough statement occurs last.

That could be patched either by removing that special casing, or adding the case_body to the list of interruptible things.

Reply:
All that paragraph says to me is that only the individual statements and compound statements can be classified as interruptions, not whether other statements are skipped or stopped.